### PR TITLE
Qt: Improve update checker system to prevent downgrades #1749

### LINKED ISSUE #1749: Improve update checker system to avoid unintended downgrades caused by potential future release patterns
### Issue

Currently, the logic is simply "if the current version number isn't equal to the latest released version number, update to the most recent release". This works in theory for most situations, however it could cause issues if we needed to do a release for a stable major release while a pre-release for a new major release is already out.

To illustrate, say we have versions 1234.1 and 1235.0-rc1, and we want to release 1234.2. The new 1234.2 update will be pushed as the most recent release for all users, regardless of their version, meaning that users on 1235.0-rc1 will be asked to "update" to 1234.2.

Because of this, we currently can't do minor releases while a future major release is in its pre-release stage, and while this hasn't been an issue until now, a planned increase of alpha prereleases for future major releases is likely to exasperate this issue to the extent that addressing it is mandatory.

To address this, we can implement code which parses the version number and checks if the installed major release number is less than or equal to the latest release before pushing an update notification. This will help to prevent unintentional downgrading.

### PR DESCRIPTION
**Pull request by RedBlackAka**
Improve update checker system to prevent downgrades. Closes #1749

If the pre-release channel is selected, the major version from the latest update is now extracted and compared the current installed version. It will only ask for updates if the new major version is higher or matches, and if the latter, then checks if there are suffix differences via the old method (e.g. allowing 2125-rc1 to upgrade to 2125-rc2), but preventing downgrades. Best way I could think of solving this and should be able to handle all edge cases

Did a quick test on my repository and seemed fine, but recommend proper testing on your end

### CODE REVIEW

**Comment by RedBlackAka** [comment]
Forgot to handle minor versions with suffixes, although is that needed? It's getting late, will have a look at it again when I can think

**Comment by OpenSauce04** [comment]
> Forgot to handle minor versions with suffixes, although is that needed? It's getting late, will have a look at it again when I can think

This is more than likely not necessary. If it turns out to be, we can handle that in another PR.

File: src/citra_qt/citra_qt.cpp [hunk 1]
```diff
@@ -180,12 +180,17 @@ bool IsPrereleaseBuild() {
...
+static bool ShouldCheckForPrereleaseUpdates() {
     const bool update_channel = UISettings::values.update_check_channel.GetValue();
     const bool using_prerelease_channel =
         (update_channel == UISettings::UpdateCheckChannels::PRERELEASE);
     return (IsPrereleaseBuild() || using_prerelease_channel);
 }
+
+static int GetMajorVersion(const std::string& version) {
+    size_t dot = version.find('.');
+    return std::stoi(version.substr(0, dot));
```
Comment (OpenSauce04) [inline src/citra_qt/citra_qt.cpp]: If the current build's version string consists entirely of letters, this causes the emulator to instantly crash on launch. While this is unlikely to happen, this shouldn't even be possible.
Please address this. I would suggest using `0` as a fallback value if the conversion isn't possible.

**Review by OpenSauce04 — CHANGES_REQUESTED** [review]

File: src/citra_qt/citra_qt.cpp (hunk 1 above)
Comment (RedBlackAka) [reply src/citra_qt/citra_qt.cpp]: Ahh I see, added a fallback to 0 if any exceptions happen. Should solve the crash, please take another look-

**Review by OpenSauce04 — APPROVED** [review]
